components/brave_rewards/resources/extension/brave_rewards/twitter.ts

```diff
@@ -129,20 +129,29 @@ const configureBraveTipAction = () => {
         const actions = tweets[i].getElementsByClassName('js-actions')[0]
         if (actions) {
           const braveTipActions = actions.getElementsByClassName('action-brave-tip')
-          if (rewards.enabled && inlineTip.enabled) {
-            if (braveTipActions.length === 0) {
-              actions.appendChild(createBraveTipAction(tweets[i]))
-            }
-          } else {
-            if (braveTipActions.length === 1) {
-              actions.removeChild(braveTipActions[0])
-            }
+          const tippingEnabled = rewards.enabled && inlineTip.enabled
+          if (tippingEnabled && braveTipActions.length === 0) {
+            actions.appendChild(createBraveTipAction(tweets[i]))
+          } else if (!tippingEnabled && braveTipActions.length === 1) {
+            actions.removeChild(braveTipActions[0])
           }
         }
       }
     })
   })
-  setTimeout(configureBraveTipAction, 3000)
 }
 
+let interval: any = null
+
+// In order to deal with infinite scrolling and overlays, periodically
+// check if injection needs to occur (mitigate the performance cost
+// by only running this when the foreground tab is active or visible)
+document.addEventListener('visibilitychange', function () {
+  if (document.hidden) {
+    clearInterval(interval)
+  } else {
+    interval = setInterval(configureBraveTipAction, 3000)
```

Comment:
I'd feel more comfortable using `setTimeout`. The problem with `setInterval` in general is that it does not wait for execution of the function to complete before scheduling the next execution. On low power devices, especially on a heavy twitter page, I imagine this function may take some time. This sets up a dangerous pattern where more cycles than necessary occur, or we have executions running back to back worst case scenario if any executions take > 3s. Even if this was unlikely, we'd be certain that this wouldn't be the cause of performance issue if we just used `setTimeout` which was scheduled at the end of each function call. You could keep the same pattern that you've got with a global variable that is cleared or setup when the document visibility is changed, just adding back the `setTimeout` from your previous commit.

Reply:
That makes sense. I didn't realize that `setInterval` had that drawback, thanks for the info on that. Just updated it as you suggested.